src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicSnippetsTransformer.java

```diff
@@ -134,14 +134,49 @@ private SnippetsFileView generateSnippets(GapicInterfaceContext context) {
       } else if (methodConfig.isPageStreaming()) {
         if (methodConfig.isFlattening()) {
           ImmutableList<FlatteningConfig> flatteningGroups = methodConfig.getFlatteningConfigs();
+          // Find flattenings that have ambiguous parameters, and mark them to use named parameters.
+          // These are page-stream flattenings that have one or two string parameters (that are not
+          // resource-names) than another flattening.
+          String[] stringParams =
+              flatteningGroups
+                  .stream()
+                  .map(
+                      flat ->
+                          flat.getFlattenedFieldConfigs()
+                              .values()
+                              .stream()
+                              .map(
+                                  field ->
+                                      field.getField().getType().isStringType()
+                                              && field.getResourceNameConfig() == null
+                                          ? 's'
+                                          : '.')
+                              .collect(
+                                  StringBuilder::new,
+                                  StringBuilder::appendCodePoint,
+                                  StringBuilder::append)
+                              .toString())
+                  .toArray(String[]::new);
```

Comment:
I have very little idea what the result is here. Is this code that exists somewhere else for another reasons?

Reply:
Added more comments to try to explain. No, this code was written here specifically for this.